hammr/commands/deploy/deploy.py

```diff
@@ -0,0 +1,152 @@
+                printer.out("Deployments:")
+                table = Texttable(800)
+                table.set_cols_dtype(["t", "t", "t", "t", "t", "t"])
+                table.header(
+                    ["Deployment name", "Deployment ID", "Hostname", "Region", "Source used", "Status"])
+                deployments = generics_utils.order_list_object_by(deployments, "name")
+                for deployment in deployments:
+                    deployment_id = deployment.applicationId
+                    deployment_status = deployment.state
+                    instances = deployment.instances.instance
+                    instance = instances[-1]
+                    if instance.scanSummary:
+                        source = "Scan ID: " + str(extract_scannedinstance_id(instance.scanSummary.uri))
+                    elif instance.applianceSummary:
+                        source = "Stack ID: " + str(generics_utils.extract_id(instance.applianceSummary.uri))
+                    else:
+                        source = str(None)
+                    if instance and instance.location and instance.hostname:
+                        table.add_row([deployment.name, deployment_id, instance.hostname, instance.location.provider, source, deployment_status])
+                    else:
+                        table.add_row([deployment.name, deployment_id, None, None, source, deployment_status])
+                print table.draw() + "\n"
+                printer.out("Found " + str(len(deployments)) + " deployments")
+
+            return 0
+        except ArgumentParserError as e:
+            printer.out("ERROR: In Arguments: " + str(e), printer.ERROR)
+            self.help_list()
+        except Exception as e:
+            return handle_uforge_exception(e)
+
+    def help_list(self):
+        doParser = self.arg_list()
+        doParser.print_help()
+
+    def arg_terminate(self):
+        doParser = ArgumentParser(prog=self.cmd_name + " terminate", add_help=True,
+                                  description="Terminate a deployment")
+        mandatory = doParser.add_argument_group("mandatory arguments")
+        mandatory.add_argument('--id', dest='id', required=True,
+                               help="id of the deployment to terminate")
+        optional = doParser.add_argument_group("optional arguments")
+        optional.add_argument('--force', '-f', dest='force', required=False, action='store_true', help='Terminate the deployment without asking for confirmation')
+        return doParser
+
+    def do_terminate(self, args):
+        try:
+            # add arguments
+            doParser = self.arg_terminate()
+            doArgs = doParser.parse_args(shlex.split(args))
+
+            if (not doArgs.force and generics_utils.query_yes_no("Do you really want to delete deployment with id '" + str(doArgs.id) + "'")) or doArgs.force:
+                status = self.api.Users(self.login).Deployments(doArgs.id).Status.Getdeploystatus()
```

Comment:
Maybe you can add a comment to explain we retrieve the status before the calling terminate to distinguish the case where the instance was already on fire from the case where it's became on_fire because of the terminate